editor/effects.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import { parse, stringify } from 'querystring';
+
+/**
+ * WordPress dependencies
+ */
+import { getBlockSettings, switchToBlockType } from 'blocks';
+
+/**
+ * Internal dependencies
+ */
+import { focusBlock, replaceBlocks } from './actions';
+
+export default {
+	REQUEST_POST_UPDATE( action, store ) {
+		const { dispatch } = store;
+		const { postId, edits } = action;
+		const isNew = ! postId;
+		const toSend = postId ? { id: postId, ...edits } : edits;
+
+		new wp.api.models.Post( toSend ).save().done( ( newPost ) => {
+			dispatch( {
+				type: 'REQUEST_POST_UPDATE_SUCCESS',
+				post: newPost,
+			} );
+		} ).fail( ( err ) => {
+			dispatch( {
+				type: 'REQUEST_POST_UPDATE_FAILURE',
+				error: get( err, 'responseJSON', {
+					code: 'unknown_error',
+					message: wp.i18n.__( 'An unknown error occurred.' ),
+				} ),
+				edits,
+				isNew,
+			} );
+		} );
+	},
+	REQUEST_POST_UPDATE_SUCCESS( action ) {
+		const { post } = action;
+		const [ baseUrl, query ] = window.location.href.split( '?' );
+		const qs = parse( query || '' );
+		if ( qs.post_id === post.id ) {
+			return;
+		}
+
+		const newUrl = baseUrl + '?' + stringify( {
+			...qs,
+			post_id: post.id,
+		} );
+		window.history.replaceState( {}, 'Post ' + post.id, newUrl );
+	},
+	TRASH_POST( action, store ) {
+		const { dispatch } = store;
+		const { postId } = action;
+		new wp.api.models.Post( { id: postId } ).destroy().done( () => {
+			dispatch( {
+				...action,
+				type: 'TRASH_POST_SUCCESS',
+			} );
+		} );
+	},
+	TRASH_POST_SUCCESS( action ) {
+		const { postId, postType } = action;
+		window.location.href = 'edit.php?' + stringify( {
+			trashed: 1,
+			post_type: postType,
+			ids: postId,
+		} );
+	},
+	MERGE_BLOCKS( action, store ) {
+		const { dispatch } = store;
+		const { blocks } = action;
+		const [ blockA, blockB ] = blocks;
```

Comment:
>Minor: `blocks` appears to only be used once in this function.  Is there an advantage to destructing then assigning?  This could be condensed to `[ blockA, blockB ] = action.blocks` with one less line.

Sure, that seems reasonable.